tasks/main.yml

```diff
@@ -4,24 +4,11 @@
     homebrew_user: '{{ homebrew_user | default(ansible_user_id) }}'
     homebrew_group: '{{ homebrew_group | default(ansible_user_gid) }}'
 
-# Homebrew setup prerequisites.
-- name: Ensure Homebrew parent directory has correct permissions (MacOS >= 10.13).
-  file:
-    path: "{{ homebrew_prefix }}"
-    owner: root
-    state: directory
-  become: yes
-  when: "ansible_distribution_version is version('10.13', '>=')"
+- include_tasks: setup-Darwin.yml
+  when: ansible_os_family == 'Darwin'
 
-- name: Ensure Homebrew parent directory has correct permissions (MacOS < 10.13).
-  file:
-    path: "{{ homebrew_prefix }}"
-    owner: root
-    group: admin
-    state: directory
-    mode: 0775
-  become: yes
-  when: "ansible_distribution_version is version('10.13', '<')"
+- include_tasks: setup-Debian.yml
```

Comment:
This should work on other Linux flavors too, right? For example, CentOS or Ubuntu?

Ideally, I want to get tests added for those platforms (though that requires restructuring the Travis CI tests a little so I can use base images like I do for my other roles like `geerlingguy.apache` for linux, versus Travis CI's macOS execution environments that I currently do, depending on the test matrix).

But doing this (naming the setup tasks `Debian` means that it wouldn't work for any other distros besides Debian (and I guess Ubuntu since it's keying on `ansible_os_family`).

Reply:
See below.